packages/eslint-plugin-nx/src/resolve-workspace-rules.ts

```diff
@@ -40,14 +41,16 @@ export const workspaceRules = ((): ESLintRules => {
     return {};
   }
   // Register `tools/eslint-rules` for TS transpilation
-  registerTSWorkspaceLint();
+  const registrationCleanup = registerTSWorkspaceLint();
   try {
     /**
      * Currently we only support applying the rules from the user's workspace plugin object
      * (i.e. not other things that plugings can expose like configs, processors etc)
      */
     const { rules } = require(WORKSPACE_PLUGIN_DIR);
-
+    if (registrationCleanup) {
```

Comment:
Thinking out loud, maybe this should come in a `finally {}` block so that it still happens in the error case? Not sure how much leakage is likely to occur with this though, so maybe it's not actually not an issue...

Reply:
It's very unlikely that registration succeeds but requiring the project fails, but I guess it's better be safe than sorry.

In the previous version we never did the cleanup at all, and we never noticed any leaks. 
I realised that by doing the cleanup we save few hundred milliseconds per run. I haven't had time to investigate why.